src/repr/src/adt/interval.rs

```diff
@@ -164,6 +164,17 @@ impl Interval {
         })
     }
 
+    /// Converts a `chrono::Duration` to an `Interval`. The resulting `Interval` will only have
+    /// microseconds, with the nanoseconds truncated. If the result exceeds the capacity of `i64`,
+    /// returns a duration of `0`. This mirrors PostgreSQL's behavior.
```

Comment:
The PR description leaves some doubt as to whether this is pg's behavior:

> I'm not sure the implementation is correct, or maybe it just follows Postgres' behavior.

But here it's asserted:

> This mirrors PostgreSQL's behavior.

Is this indeed pg's behavior?

Reply:
Looking at Postgres' `timestamp_mi`, which seems to be the equivalent, there's a [check](https://doxygen.postgresql.org/backend_2utils_2adt_2timestamp_8c.html#abaecae5fa010fe1bbb6ed80e5508afe7) that the result fits. I changed the PR to do the same.